client/login/desktop-login/index.tsx

```diff
@@ -0,0 +1,115 @@
+import './style.scss';
+import { Button } from '@wordpress/components';
+import debugFactory from 'debug';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import Main from 'calypso/components/main';
+import { Notice } from 'calypso/components/notice';
+import WpcomLoginForm from 'calypso/signup/wpcom-login-form';
+
+const debug = debugFactory( 'calypso:desktop' );
+
+// The desktop app will intercept this URL and start the login in the user's external browser.
+const loginUrl = '/desktop-start-login';
+
+type Props =
+	| { action: 'start'; accessToken?: never; error?: never }
```

Comment:
Isn't it more correct when the `start` type is just `{ action: 'start' }`? Then you'll to reorganize the `DesktopLogin` code to guard access to `accessToken` and `error` props with `isStart` and `isFinalize` condition. And that looks like an improvement that clarifies the code.